src/bp/ui-shared/src/TokenRefresher/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { StoredToken } from 'common/typings'
+
+import ms from 'ms'
+import { FC, useEffect, useState } from 'react'
+
+import { TokenRefresherProps } from './typings'
+import { tokenNeedsRefresh, getToken, setToken } from '~/../../ui-shared-lite/auth'
+
+const REFRESH_INTERVAL = ms('5m')
+
+const TokenRefresher: FC<TokenRefresherProps> = props => {
```

Comment:
It's kind of weird to hide pure javascript logic in a component just because we're in a react app .. IMO components should mostly be for rendering interactive html

Reply:
Yeah, I just wanted to avoid having a different logic for the admin and the studio, since the refresh logic is the same. I guess we'll change this in the future !